src/host/screenInfo.cpp

```diff
@@ -1462,12 +1464,17 @@ bool SCREEN_INFORMATION::IsMaximizedY() const
 
     if (SUCCEEDED(hr))
     {
-        auto& newCursor = newTextBuffer->GetCursor();
+        // Make sure the new virtual bottom is far enough down to include both
+        // the cursor row and the last non-space row.
+        const auto cursorRow = newTextBuffer->GetCursor().GetPosition().Y;
+        const auto lastNonSpaceRow = newTextBuffer->GetLastNonSpaceCharacter().Y;

```

Comment:
Yeah. I would have thought the `TextBuffer::Reflow` should have been able to provide that information for us, since it's just created that new buffer. But then that made me ask why the `TextBuffer` isn't tracking the virtual bottom in general? And why we've rewritten essentially the same resizing code in the `Terminal` class but in a completely different way? (Although it's worth noting that the `Terminal` implementation also uses `GetLastNonSpaceCharacter`.) And my conclusion was that all of this buffer management code really needs a unified rewrite, but that was way more effort than I wanted to get into now, if ever. Hence the quick fix hack.

Reply:
That's mostly on me. SCREEN_INFO is pretty tightly bound to the rest of the _console_ itself. It was a lot harder to separate out from... everything else that makes the console _the console_. I was trying to create a version with a cleaner interface in `Terminal`.

In retrospect, yes, a unified implementation would have been smart. I think I would have liked to create a cleaner boundary around SCREEN_INFO so that terminal could have reused it too.

Reply:
Yeah, I realise it won't be easy decoupling that code from the console, but I think it'll be worth revisiting at some point, possibly when/if we get around to implementing VT paging. Because ideally I'd like for us to have a unified buffer management class that can handle both the console buffers and the VT pages in the same way (along with XTerm's alt/main weirdness). And that class could hopefully then be shared between conhost and Terminal and reduce some of the duplication. But it's not urgent.